docs/source/checks/tabular/performance/plot_model_error_analysis.py

```diff
@@ -2,47 +2,80 @@
+This notebooks provides an overview for using and understanding the model error analysis check.
 
-from urllib.request import urlopen
+**Structure:**
 
-import pandas as pd
+* `What is Model Error Analysis? <#what-is-model-error-analysis>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
 
-from deepchecks.tabular.datasets.classification import adult
+What is Model Error Analysis?
+=============================
+Looking at performance metrics are often enough when comparing various modeling alternatives or when tracking model
+progress during training. However, when it's time to fully evaluate if a model is fit for production, it's recommended
+to look deeper at how the model performs on various segments of the data. The model error analysis check searches for
+data segments in which the model error is significantly lower from the model error of the dataset as a whole.
 
-#%%
-# Create Dataset
-# ==============
+Algorithm:
+----------
 
-train_ds, test_ds = adult.load_data()
+1. Computes the per-sample loss (for log-loss for classification, mse for regression).
+2. Trains a regression model to predict the error of the user's model, based on the input features.
+3. Do stage 3 several times with various tree parameters and random states to ensure that the most relevant partitions for model error are selected.
+4. The features scoring the highest feature importance for the error regression model are selected and the distribution of the error vs the feature values is plotted.
 
-#%%
-# Classification Model
-# ====================
+The check results are shown only if the error regression model manages
+to predict the error well enough. The resulting plots show the distribution of the error for the features that are
+most effective at segmenting the error to high and low values, without need for manual selection of segmentation
+features.
 
-model = adult.load_fitted_model()
+Related Checks:
+---------------
+When the important segments of the data are known in advance (when we know that some population segments have
+different behaviours and business importance, for example income levels or state of residence) it is possible to just
+have a look at the performance at various pre-defined segments. In deepchecks, this can be done using the
+:doc:`Segment Performance </checks_gallery/tabular/performance/plot_segment_performance>`_ check, which shows the
+performance for segments defined by combination of values from two pre-defined columns.
 
-#%%
-# Run Check
-# =========
 
+Run the check
+=============
+We will run the check on the adult dataset which can be downloaded from the
+`UCI machine learning repository <http://archive.ics.uci.edu/ml>`_ and is also available in
+`deepchecks.tabular.datasets`.
+"""
+
+from deepchecks.tabular.datasets.classification import adult
 from deepchecks.tabular.checks import ModelErrorAnalysis
 
-#%%
+train_ds, test_ds = adult.load_data()
```

Comment:
```suggestion
train_ds, test_ds = adult.load_data(data_format='Dataset', as_train_test=True)
```

Comment - think it's better to make these explicit in examples so that when users try to run it on their data it increases the chances they understand they should have a "Dataset"